dynamic_programming/longest_common_string.cpp

```diff
@@ -1,53 +1,142 @@
-#include <iosrteam>
-using namespace std;
+/**
+ * @file
+ * @brief contains the definition of the function longest_common_string_length
+ * @details
+ * the function longest_common_string_length computes the length
+ * of the longest common string which can be created of two input strings
+ * by removing characters from them
+ */
 
-int max(int a, int b) { return (a > b) ? a : b; }
+#include <cassert>  /// for assert
+#include <string>   /// for std::string
+#include <utility>  /// for std::move
+#include <vector>   /// for std::vector
 
-int main() {
-    char str1[] = "DEFBCD";
-    char str2[] = "ABDEFJ";
-    int i, j, k;
-    int n = strlen(str1) + 1;
-    int m = strlen(str2) + 1;
-    // cout<<n<<" "<<m<<"\n";
-    int a[m][n];
-
-    for (i = 0; i < m; i++) {
-        for (j = 0; j < n; j++) {
-            if (i == 0 || j == 0)
-                a[i][j] = 0;
-
-            else if (str1[i - 1] == str2[j - 1])
-                a[i][j] = a[i - 1][j - 1] + 1;
-
-            else
-                a[i][j] = 0;
-        }
-    }
+/**
+ * @brief computes the length of the longest common string created from input
+ * strings
+ * @details has O(str_a.size()*str_b.size()) time and memory complexity
+ * @param str_a first input string
+ * @param str_b second input string
+ * @returns the length of the longest common string which can be strated from
+ * str_a and str_b
+ */
+std::size_t longest_common_string_length(const std::string& str_a,
+                                         const std::string& str_b) {
```

Comment:
Could you expand the parameter names to be more descriptive? 
Ex change str_a to string_a
I understand that str is just a short form but we are not short on time